helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -150,12 +150,22 @@ public String getDomainAsString() {
    */
   public Map<String, String> getDomainAsMap() {
     String domain = getDomainAsString();
+    Map<String, String> domainAsMap = new HashMap<>();
     if (domain == null || domain.isEmpty()) {
-      return Collections.emptyMap();
+      return domainAsMap;
+    }
+
+    String[] pathPairs = domain.trim().split(",");
+    for (String pair : pathPairs) {
+      String[] values = pair.trim().split("=");
+      if (values.length != 2 || values[0].isEmpty() || values[1].isEmpty()) {
+        throw new IllegalArgumentException(
+            String.format("Domain-Value pair %s is not valid.", pair));
+      }
+      domainAsMap.put(values[0], values[1]);
```

Comment:
We shall trim at the final values level but not the above layers. What do you think?

Reply:
I think if the value or key is null, then this pair shouldn't be count as a valid input. 
Since currently the callers of this function only checks if key exists in the returned map and uses the corresponding value. So i think the user all assume all k-v pairs in  returned map are valid. 
Thus, I think we should throw exception for empty value. (Original version already handles no splitter and null key errors).

Reply:
Consider this case:
"a = b, c = d", the current logic will have a map of {"a ": " b", "c " : " d"} (notice the space), and what we expect is {"a": "b", "c" : "d"}.

Reply:
Oh I got what you are saying. Yes thats correct. Will update.